File: sdks/java/src/test/java/com/hellosign/openapi/api/AccountApiTest.java
package com.hellosign.openapi.api;

import com.hellosign.openapi.ApiClient;
import com.hellosign.openapi.ApiException;
import com.hellosign.openapi.TestHelper;
import com.hellosign.openapi.model.*;
import org.junit.Assert;
import org.junit.Test;

import java.util.stream.IntStream;

public class AccountApiTest {
    @Test
    public void accountCreateTest() throws Exception {
        AccountCreateRequest request = TestHelper.getFixtureData(
            AccountCreateRequest.class,
            "default"
        );
        AccountCreateResponse expected = TestHelper.getFixtureData(
            AccountCreateResponse.class,
            "default"
        );
        ApiClient apiClient = TestHelper.setUpMock(200, expected);

        AccountApi accountApi = new AccountApi(apiClient);
        AccountCreateResponse actual = accountApi.accountCreate(request);
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void accountGetTest() throws Exception {
        AccountGetResponse expected = TestHelper.getFixtureData(
            AccountGetResponse.class,
            "default"
        );
        ApiClient apiClient = TestHelper.setUpMock(200, expected);

        AccountApi accountApi = new AccountApi(apiClient);
        AccountGetResponse actual = accountApi.accountGet(null, "jack@example.com");
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void accountUpdateTest() throws Exception {
        AccountUpdateRequest request = TestHelper.getFixtureData(
            AccountUpdateRequest.class,
            "default"
        );
        AccountGetResponse expected = TestHelper.getFixtureData(
            AccountGetResponse.class,
            "default"
        );
        ApiClient apiClient = TestHelper.setUpMock(200, expected);

        AccountApi accountApi = new AccountApi(apiClient);
        AccountGetResponse actual = accountApi.accountUpdate(request);
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void accountVerifyTest() throws Exception {
        AccountVerifyRequest request = TestHelper.getFixtureData(
            AccountVerifyRequest.class,
            "default"
        );
        AccountVerifyResponse expected = TestHelper.getFixtureData(
            AccountVerifyResponse.class,
            "default"
        );
        ApiClient apiClient = TestHelper.setUpMock(200, expected);

        AccountApi accountApi = new AccountApi(apiClient);
        AccountVerifyResponse actual = accountApi.accountVerify(request);
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void testHttpCodeRange() throws Exception {
        AccountVerifyRequest request = TestHelper.getFixtureData(
            AccountVerifyRequest.class,
            "default"
        );
        ErrorResponse expected = TestHelper.getFixtureData(
            ErrorResponse.class,
            "default"
        );

        IntStream.range(400, 500).forEach(value -> {
            try {
                ApiClient apiClient = TestHelper.setUpMock(value, expected);
                AccountApi accountApi = new AccountApi(apiClient);
                accountApi.accountVerify(request);
                Assert.fail();
            } catch (ApiException e) {
                Assert.assertEquals(value, e.getCode());
                Assert.assertEquals(expected, e.getErrorResponse());
            }
        });
    }
}

File: sdks/node/tests/api/accountApi.test.ts
import 'jest';

import { AccountApi } from '../../api/';
import * as m from '../../model/';
import {
  getFixtureData,
  setExpectedResponse,
  diffJson,
} from '../test_utils';

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

describe('AccountApiTest', () => {
  let mock: typeof MockAdapter;

  beforeEach(() => {
    mock = new MockAdapter(axios);
  });

  const api = new AccountApi();

  it('testHttpCodeRange', () => {
    const requestClass = 'AccountVerifyRequest';
    const requestData = getFixtureData(requestClass)['default'];

    const responseClass = 'ErrorResponse';
    const responseData = getFixtureData(responseClass)['default'];

    const code = Math.round(Math.random() * (499 - 400)) + 400;
    setExpectedResponse(mock, responseData, code);

    const obj = m.AccountVerifyRequest.init(requestData);

    api.accountVerify(obj).then(response => {
      throw new Error('Should have thrown error: ' + response);
    }).catch(error => {
      expect(error.body.constructor.name).toBe(responseClass);
    });
  });

  it('testCamelCaseGeneratesCorrectRequestPayload', () => {
    const userRequestData = {
      emailAddress: "newuser@dropboxsign.com",
      clientId: "cc91c61d00f8bb2ece1428035716b",
      clientSecret: "1d14434088507ffa390e6f5528465"
    };

    const actualRequestData = m.AccountCreateRequest.init({
      email_address: "newuser@dropboxsign.com",
      client_id: "cc91c61d00f8bb2ece1428035716b",
      client_secret: "1d14434088507ffa390e6f5528465"
    });

    setExpectedResponse(mock, {}, 200, undefined, actualRequestData);
    api.accountCreate(userRequestData);
  });

  it('testAccountCreate', () => {
    const requestClass = 'AccountCreateRequest';
    const requestData = getFixtureData(requestClass)['default'];

    const responseClass = 'AccountCreateResponse';
    const responseData = getFixtureData(responseClass)['default'];

    setExpectedResponse(mock, responseData, 200);

    const obj = m.AccountCreateRequest.init(requestData);

    api.accountCreate(obj).then(response => {
      const diff = diffJson(
        response.body,
        m.AccountCreateResponse.init(responseData),
      );

      expect(response.body.constructor.name).toBe(responseClass);
      expect(diff).toBeFalsy();
    }).catch(error => {
      throw new Error('Should not have thrown error: ' + error);
    });
  });

  it('testAccountGet', () => {
    const responseClass = 'AccountGetResponse';
    const responseData = getFixtureData(responseClass)['default'];

    setExpectedResponse(mock, responseData, 200);

    api.accountGet(undefined, "jack@example.com").then(response => {
      const diff = diffJson(
        response.body,
        m.AccountGetResponse.init(responseData),
      );

      expect(response.body.constructor.name).toBe(responseClass);
      expect(diff).toBeFalsy();
    }).catch(error => {
      throw new Error('Should not have thrown error: ' + error);
    });
  });

  it('testAccountUpdate', () => {
    const requestClass = 'AccountUpdateRequest';
    const requestData = getFixtureData(requestClass)['default'];

    const responseClass = 'AccountGetResponse';
    const responseData = getFixtureData(responseClass)['default'];

    setExpectedResponse(mock, responseData, 200);

    const obj = m.AccountUpdateRequest.init(requestData);

    api.accountUpdate(obj).then(response => {
      const diff = diffJson(
        response.body,
        m.AccountGetResponse.init(responseData),
      );

      expect(response.body.constructor.name).toBe(responseClass);
      expect(diff).toBeFalsy();
    }).catch(error => {
      throw new Error('Should not have thrown error: ' + error);
    });
  });

  it('testAccountVerify', () => {
    const requestClass = 'AccountVerifyRequest';
    const requestData = getFixtureData(requestClass)['default'];

    const responseClass = 'AccountVerifyResponse';
    const responseData = getFixtureData(responseClass)['default'];

    setExpectedResponse(mock, responseData, 200);

    const obj = m.AccountVerifyRequest.init(requestData);

    api.accountVerify(obj).then(response => {
      const diff = diffJson(
        response.body,
        m.AccountVerifyResponse.init(responseData),
      );

      expect(response.body.constructor.name).toBe(responseClass);
      expect(diff).toBeFalsy();
    }).catch(error => {
      throw new Error('Should not have thrown error: ' + error);
    });
  });
});
